kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/SolverServicesClientImpl.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.server.client.impl;
+
+import org.kie.api.command.Command;
+import org.kie.server.api.commands.CallContainerCommand;
+import org.kie.server.api.commands.CommandScript;
+import org.kie.server.api.model.KieServerCommand;
+import org.kie.server.api.model.ServiceResponse;
+import org.kie.server.api.model.instance.SolverInstance;
+import org.kie.server.api.model.instance.SolverInstanceList;
+import org.kie.server.api.rest.RestURI;
+import org.kie.server.client.KieServicesConfiguration;
+import org.kie.server.client.RuleServicesClient;
+import org.kie.server.client.SolverServicesClient;
+import org.optaplanner.core.api.domain.solution.Solution;
+
+import java.util.Collections;
+
+public class SolverServicesClientImpl
+        extends AbstractKieServicesClientImpl implements SolverServicesClient {
+
+    public SolverServicesClientImpl(KieServicesConfiguration config) {
+        super(config);
+    }
+
+    public SolverServicesClientImpl(KieServicesConfiguration config, ClassLoader classLoader) {
+        super(config, classLoader);
+    }
+
+    @Override
+    public ServiceResponse<SolverInstanceList> getSolvers(String containerId) {
+        checkMandatoryParameter( "ContainerID", containerId );
+        if( config.isRest() ) {
+            String uri = getURI( containerId );
+            return makeHttpGetRequestAndCreateServiceResponse( uri, SolverInstanceList.class );
+        } else {
+            //            CommandScript script = new CommandScript( Collections.singletonList((KieServerCommand) new CallContainerCommand(id, payload)) );
+            //            return (ServiceResponse<String>) executeJmsCommand( script ).getResponses().get( 0 );
+            return null;
```

Comment:
maybe more meaningful would be to throw UnsupportedOperationException as null might get people worried that it does not work while it's simply not supported/implemented